apps/vs-code-designer/src/app/commands/createNewCodeProject/cloudToLocal.ts

```diff
@@ -0,0 +1,199 @@
+import { window } from 'vscode';
+import * as path from 'path';
+import AdmZip = require('adm-zip');
+import * as fs from 'fs';
+import { SetLogicAppType } from './CodeProjectBase/setLogicAppType';
+import { writeFormattedJson } from '../../utils/fs';
+
+// Constants
+const openFolder = true;
+
+// Function to create an AdmZip instance
+function createAdmZipInstance(zipFilePath: string) {
+  return new AdmZip(zipFilePath);
+}
+
+// Function to get zip entries
+function getZipEntries(zipFilePath: string) {
+  const zip = createAdmZipInstance(zipFilePath);
+  return zip.getEntries();
+}
+
+async function cleanLocalSettings(localSettingsPath: string) {
+  const localSettings = JSON.parse(fs.readFileSync(localSettingsPath, 'utf8'));
+
+  if (localSettings.Values) {
+    Object.keys(localSettings.Values).forEach((key) => {
+      if (
+        key === 'WEBSITE_SITE_NAME' ||
+        key === 'WEBSITE_AUTH_ENABLED' ||
+        key === 'WEBSITE_SLOT_NAME' ||
+        key === 'ScmType' ||
+        key === 'FUNCTIONS_RUNTIME_SCALE_MONITORING_ENABLED'
+      ) {
+        delete localSettings.Values[key];
+      } else if (key === 'AzureWebJobsStorage') {
+        localSettings.Values[key] = 'UseDevelopmentStorage=true';
+      }
+    });
+
+    await writeFormattedJson(localSettingsPath, localSettings);
+  }
+}
+
+// Main function to orchestrate the cloud to local process
+export async function cloudToLocalInternal(
+  context: IActionContext,
+  options: ICreateFunctionOptions = {
+    folderPath: undefined,
+    language: undefined,
+    version: undefined,
+    templateId: undefined,
+    functionName: undefined,
+    functionSettings: undefined,
+    suppressOpenFolder: !openFolder,
+  }
+): Promise<void> {
+  addLocalFuncTelemetry(context);
+  showPreviewWarning(extensionCommand.cloudToLocal);
+
+  const language: ProjectLanguage | string = (options.language as ProjectLanguage) || getGlobalSetting(projectLanguageSetting);
+  const version: string = options.version || getGlobalSetting(funcVersionSetting) || (await tryGetLocalFuncVersion()) || latestGAVersion;
+  const projectTemplateKey: string | undefined = getGlobalSetting(projectTemplateKeySetting);
+  const wizardContext: Partial<IFunctionWizardContext> & IActionContext = Object.assign(context, options, {
+    language,
+    version: tryParseFuncVersion(version),
+    projectTemplateKey,
+    projectPath: options.folderPath,
+  });
+
+  if (options.suppressOpenFolder) {
+    wizardContext.openBehavior = OpenBehavior.dontOpen;
+  } else if (!wizardContext.openBehavior) {
+    wizardContext.openBehavior = getWorkspaceSetting(projectOpenBehaviorSetting);
+    context.telemetry.properties.openBehaviorFromSetting = String(!!wizardContext.openBehavior);
+  }
+
+  const wizard: AzureWizard<IFunctionWizardContext> = new AzureWizard(wizardContext, {
+    title: localize('cloudToLocal', 'Import zip into new Workspace'),
```

Comment:
localize first string (id) should match or at least make some reference to the actual localized text. `Workspace` should not be with upper case letter